x-pack/filebeat/input/gcs/pool/pool.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package pool
+
+import (
+	"context"
+	"sync"
+
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/job"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type Pool interface {
```

Comment:
Please add docs for this and its methods.

Looking through the code, I think it would be better to have a limiter and spin up goroutines as needed rather than have a worker pool.

Reply:
Sure will add the docs. As for having a limiter and spinning go routines up, this approach was made to stay in line with the other inputs such as s3, office365 etc. They provide a means for the user to configure the max number of workers in the config.

Reply:
A limiter does this too, but is simpler since it uses the runtime to do all the work that is done here.